src/main/java/gyro/google/compute/SecurityPolicyRule.java

```diff
@@ -0,0 +1,196 @@
+    public void setPriority(Integer priority) {
+        this.priority = priority;
+    }
+
+    /**
+     * The action to take for this rule. (Required)
+     */
+    @Updatable
+    @Required
+    @ValidStrings({ "allow", "deny(403)", "deny(404)", "deny(502)" })
+    public String getAction() {
+        return action;
+    }
+
+    public void setAction(String action) {
+        this.action = action;
+    }
+
+    /**
+     * The preview flag indicates that this rule is not enforced.
+     */
+    @Updatable
+    public Boolean getPreview() {
+        return preview;
+    }
+
+    public void setPreview(Boolean preview) {
+        this.preview = preview;
+    }
+
+    /**
+     * The match for this rule. (Required).
+     */
+    @Updatable
+    @Required
+    public SecurityPolicyRuleMatcher getMatch() {
+        return match;
+    }
+
+    public void setMatch(SecurityPolicyRuleMatcher match) {
+        this.match = match;
+    }
+
+    @Override
+    public String primaryKey() {
+        return "security-policy-rule-" + getPriority();
+    }
+
+    com.google.api.services.compute.model.SecurityPolicyRule toSecurityPolicyRule() {
+        com.google.api.services.compute.model.SecurityPolicyRule policyRule = new com.google.api.services.compute.model.SecurityPolicyRule();
+        policyRule.setAction(getAction());
+        policyRule.setDescription(getDescription());
+        policyRule.setPriority(getPriority());
+        policyRule.setPreview(getPreview());
+
+        if (getMatch() != null) {
+            policyRule.setMatch(match.toSecurityPolicyRuleMatcher());
+        }
+
+        return policyRule;
+    }
+
+    @Override
+    public void copyFrom(com.google.api.services.compute.model.SecurityPolicyRule securityPolicyRule) {
+        setPriority(securityPolicyRule.getPriority());
+        setDescription(securityPolicyRule.getDescription());
+        setAction(securityPolicyRule.getAction());
+        setPreview(securityPolicyRule.getPreview());
+
+        if (securityPolicyRule.getMatch() != null) {
+            SecurityPolicyRuleMatcher matcher = newSubresource(SecurityPolicyRuleMatcher.class);
+            matcher.copyFrom(securityPolicyRule.getMatch());
+            setMatch(matcher);
+        }
+    }
+
+    @Override
+    protected boolean doRefresh() throws Exception {
+        return true;
+    }
+
+    @Override
+    protected void doCreate(GyroUI ui, State state) throws Exception {
+        Compute client = createComputeClient();
+
+        SecurityPolicyResource securityPolicyResource = (SecurityPolicyResource) this.parentResource();
+        Operation operation = client.securityPolicies()
+            .addRule(getProjectId(), securityPolicyResource.getName(), toSecurityPolicyRule())
+            .execute();
+        waitForCompletion(client, operation);
+
+        refresh();
+    }
+
+    @Override
+    public void doUpdate(
+        GyroUI ui, State state, Resource current, Set<String> changedFieldNames) throws Exception {
```

Comment:
if the update only handles updating `description`, `preview` and `priority` why action and rule matcher are updatable fields ?